include/extractor/guidance/intersection_generator.hpp

```diff
@@ -22,6 +22,13 @@ namespace extractor
 {
 namespace guidance
 {
+
+struct IntersectionGenerationParameters
+{
+    NodeID nid;
+    EdgeID via_eid;
+};
```

Comment:
Hm yeah I think my main issue is with how we handle nodes and edges, and compositions of them. They're completely disconnected from the type system. That is, I can
- pass an edge id where a node id is required (they're both weak uint32_t aliases)
- pass a node id for a completely different intersection not corresponding to the via edge id

when in reality node ids should be typed and be distinguishable to edge ids at compile time. And starting node and via edge should probably, too. Then all our guidance code could work on those.

But I guess this is way out of scope for this pull request ..